Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docrepository/adapter/AdapterComponentDocRepositoryOrchImpl.java

```diff
@@ -520,8 +522,11 @@ protected DocumentMetadata setDocument(
             try {
                 DataHandler dh = docMap.get(extrinsicObject.getId());
                 rawData = getLargeFileUtils().convertToBytes(dh);
-                Document document = new Document(doc);
+                Document document = new Document();
                 document.setRawData(rawData);
+                document.setRepositoryUniqueId(getDefaultRepositoryId());
```

Comment:
Not sure I understand these changes. The DocumentMetadata was failing to have the `newRepositoryUniqueId` variable set, so we are setting it on the document only, and not the metadata? Why? Shouldnt the real solution be to populate the `newRepositoryUniqueId` field on the metadata and propogate that over instead? `newRepositoryUniqueId` is essentially a foreign key belonging to Document for later lookup.

Reply:
NewRepositoryUniqueId is required when on-demand document present.  Some documents are not on-demand.